Upstream-Status: Backport

Signed-off-by: Roy.Li <rongqing.li@windriver.com> 


From 597a7348527cbe91e35cf04a2e214a562612a768 Mon Sep 17 00:00:00 2001
From: YunQiang Su <wzssyqa@gmail.com>
Date: Sat, 22 Jun 2013 06:14:09 +0200
Subject: [PATCH] Add support for mipsn32(el) and mips64(el) to arch tables

[guillem@debian.org:
 - Remove unneeded abi entry.
 - Fix mips64 endianness. ]

Closes: #685096, #707323

Signed-off-by: Guillem Jover <guillem@debian.org>
---
 abitable         |    1 +
 cputable         |    2 ++
 ostable          |    2 ++
 triplettable     |    4 ++++
 5 files changed, 11 insertions(+)

diff --git a/abitable b/abitable
index 578228d..300a1ab 100644
--- a/abitable
+++ b/abitable
@@ -7,4 +7,5 @@
 # Column 2 is the size (in bits) of the ABI integers/pointers.
 #
 # <Debian name>	<Bits>
+gnuabin32	32
 gnux32		32
diff --git a/cputable b/cputable
index c556fdd..d48e59c 100644
--- a/cputable
+++ b/cputable
@@ -27,6 +27,8 @@ m32r		m32r		m32r			32	big
 m68k		m68k		m68k			32	big
 mips		mips		mips(eb)?		32	big
 mipsel		mipsel		mipsel			32	little
+mips64		mips64		mips64			64	big
+mips64el	mips64el	mips64el		64	little
 powerpc		powerpc		(powerpc|ppc)		32	big
 ppc64		powerpc64	(powerpc|ppc)64		64	big
 s390		s390		s390			32	big
diff --git a/ostable b/ostable
index 1383150..29d3843 100644
--- a/ostable
+++ b/ostable
@@ -18,6 +18,8 @@ musleabihf-linux	linux-musleabihf	linux[^-]*-musleabihf
 musl-linux		linux-musl		linux[^-]*-musl
 gnueabihf-linux		linux-gnueabihf		linux[^-]*-gnueabihf
 gnueabi-linux		linux-gnueabi		linux[^-]*-gnueabi
+gnuabin32-linux		linux-gnuabin32		linux[^-]*-gnuabin32
+gnuabi64-linux		linux-gnuabi64		linux[^-]*-gnuabi64
 gnuspe-linux		linux-gnuspe		linux[^-]*-gnuspe
 gnux32-linux		linux-gnux32		linux[^-]*-gnux32
 gnulp-linux		linux-gnulp		linux[^-]*-gnulp
diff --git a/triplettable b/triplettable
index e0d78a4..a2c683f 100644
--- a/triplettable
+++ b/triplettable
@@ -9,6 +9,10 @@ musleabihf-linux-arm	musl-linux-armhf
 musl-linux-<cpu>	musl-linux-<cpu>
 gnueabihf-linux-arm	armhf
 gnueabi-linux-arm	armel
+gnuabin32-linux-mips64el	mipsn32el
+gnuabin32-linux-mips64	mipsn32
+gnuabi64-linux-mips64el	mips64el
+gnuabi64-linux-mips64	mips64
 gnuspe-linux-powerpc	powerpcspe
 gnux32-linux-amd64	x32
 gnulp-linux-i386	lpia
-- 
1.7.10.4

